lib/src/main/kotlin/com/swmansion/starknet/service/http/HttpRequest.kt

```diff
@@ -1,23 +1,81 @@
 package com.swmansion.starknet.service.http
 
 import com.swmansion.starknet.provider.Request
+import com.swmansion.starknet.provider.rpc.JsonRpcRequest
+import com.swmansion.starknet.provider.rpc.buildJsonBatchHttpDeserializer
+import com.swmansion.starknet.provider.rpc.buildJsonHttpDeserializer
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.json.Json
 import java.util.concurrent.CompletableFuture
 import java.util.function.Function
 
 typealias HttpResponseDeserializer<T> = Function<HttpResponse, T>
 
 class HttpRequest<T>(
-    val payload: HttpService.Payload,
-    val deserializer: HttpResponseDeserializer<T>,
+    val url: String,
+    val jsonRpcRequest: JsonRpcRequest,
+    val serializer: KSerializer<T>,
+    val deserializationJson: Json,
     val service: HttpService,
 ) : Request<T> {
 
+    private fun getPayload(): HttpService.Payload {
+        val body = Json.encodeToString(JsonRpcRequest.serializer(), jsonRpcRequest)
```

Comment:
Nit:
```suggestion
        val body = Json.encodeToString(jsonRpcRequest)
```
we can also inline it